--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,459 @@
+load("@rules_erlang//:erlang_bytecode2.bzl", "erlang_bytecode")
+load("@rules_erlang//:app_file.bzl", "app_file")
+load("@rules_erlang//:erlang_app_info.bzl", "erlang_app_info")
+load("@rules_erlang//:untar.bzl", "untar")
+
+untar(
+    name = "contents",
+    outs = [
+        "LICENSE",
+        "Makefile",
+        "README.asciidoc",
+        "erlang.mk",
+        "include/cow_inline.hrl",
+        "include/cow_parse.hrl",
+        "src/cow_base64url.erl",
+        "src/cow_cookie.erl",
+        "src/cow_date.erl",
+        "src/cow_hpack.erl",
+        "src/cow_hpack_dec_huffman_lookup.hrl",
+        "src/cow_http.erl",
+        "src/cow_http2.erl",
+        "src/cow_http2_machine.erl",
+        "src/cow_http_hd.erl",
+        "src/cow_http_struct_hd.erl",
+        "src/cow_http_te.erl",
+        "src/cow_iolists.erl",
+        "src/cow_link.erl",
+        "src/cow_mimetypes.erl",
+        "src/cow_mimetypes.erl.src",
+        "src/cow_multipart.erl",
+        "src/cow_qs.erl",
+        "src/cow_spdy.erl",
+        "src/cow_spdy.hrl",
+        "src/cow_sse.erl",
+        "src/cow_uri.erl",
+        "src/cow_uri_template.erl",
+        "src/cow_ws.erl",
+        "src/cowlib.app.src",
+    ],
+    archive = "contents.tar.gz",
+)
+
+erlang_bytecode(
+    name = "beam_files_0",
+    srcs = ["src/cow_base64url.erl"],
+    outs = ["ebin/cow_base64url.beam"],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_1",
+    srcs = ["src/cow_cookie.erl"],
+    outs = ["ebin/cow_cookie.beam"],
+    hdrs = ["include/cow_inline.hrl"],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_2",
+    srcs = ["src/cow_date.erl"],
+    outs = ["ebin/cow_date.beam"],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_3",
+    srcs = ["src/cow_hpack.erl"],
+    outs = ["ebin/cow_hpack.beam"],
+    hdrs = ["src/cow_hpack_dec_huffman_lookup.hrl"],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_4",
+    srcs = ["src/cow_http.erl"],
+    outs = ["ebin/cow_http.beam"],
+    hdrs = ["include/cow_inline.hrl"],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_5",
+    srcs = ["src/cow_http2.erl"],
+    outs = ["ebin/cow_http2.beam"],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_6",
+    srcs = ["src/cow_http2_machine.erl"],
+    outs = ["ebin/cow_http2_machine.beam"],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_7",
+    srcs = ["src/cow_http_hd.erl"],
+    outs = ["ebin/cow_http_hd.beam"],
+    hdrs = [
+        "include/cow_inline.hrl",
+        "include/cow_parse.hrl",
+    ],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_8",
+    srcs = ["src/cow_http_struct_hd.erl"],
+    outs = ["ebin/cow_http_struct_hd.beam"],
+    hdrs = ["include/cow_parse.hrl"],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_9",
+    srcs = ["src/cow_http_te.erl"],
+    outs = ["ebin/cow_http_te.beam"],
+    hdrs = ["include/cow_parse.hrl"],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_10",
+    srcs = ["src/cow_iolists.erl"],
+    outs = ["ebin/cow_iolists.beam"],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_11",
+    srcs = ["src/cow_link.erl"],
+    outs = ["ebin/cow_link.beam"],
+    hdrs = [
+        "include/cow_inline.hrl",
+        "include/cow_parse.hrl",
+    ],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_12",
+    srcs = ["src/cow_mimetypes.erl"],
+    outs = ["ebin/cow_mimetypes.beam"],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_13",
+    srcs = ["src/cow_multipart.erl"],
+    outs = ["ebin/cow_multipart.beam"],
+    hdrs = ["include/cow_inline.hrl"],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_14",
+    srcs = ["src/cow_qs.erl"],
+    outs = ["ebin/cow_qs.beam"],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_15",
+    srcs = ["src/cow_spdy.erl"],
+    outs = ["ebin/cow_spdy.beam"],
+    hdrs = ["src/cow_spdy.hrl"],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_16",
+    srcs = ["src/cow_sse.erl"],
+    outs = ["ebin/cow_sse.beam"],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_17",
+    srcs = ["src/cow_uri.erl"],
+    outs = ["ebin/cow_uri.beam"],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_18",
+    srcs = ["src/cow_uri_template.erl"],
+    outs = ["ebin/cow_uri_template.beam"],
+    hdrs = [
+        "include/cow_inline.hrl",
+        "include/cow_parse.hrl",
+    ],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+erlang_bytecode(
+    name = "beam_files_19",
+    srcs = ["src/cow_ws.erl"],
+    outs = ["ebin/cow_ws.beam"],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+deterministic",
+        ],
+    }),
+)
+
+app_file(
+    name = "app_file",
+    app_description = "Support library for manipulating Web protocols.",
+    app_name = "cowlib",
+    app_src = ["src/cowlib.app.src"],
+    app_version = "2.9.1",
+    dest = "ebin",
+    modules = [
+        "ebin/cow_base64url.beam",
+        "ebin/cow_cookie.beam",
+        "ebin/cow_date.beam",
+        "ebin/cow_hpack.beam",
+        "ebin/cow_http.beam",
+        "ebin/cow_http2.beam",
+        "ebin/cow_http2_machine.beam",
+        "ebin/cow_http_hd.beam",
+        "ebin/cow_http_struct_hd.beam",
+        "ebin/cow_http_te.beam",
+        "ebin/cow_iolists.beam",
+        "ebin/cow_link.beam",
+        "ebin/cow_mimetypes.beam",
+        "ebin/cow_multipart.beam",
+        "ebin/cow_qs.beam",
+        "ebin/cow_spdy.beam",
+        "ebin/cow_sse.beam",
+        "ebin/cow_uri.beam",
+        "ebin/cow_uri_template.beam",
+        "ebin/cow_ws.beam",
+    ],
+    stamp = 0,
+)
+
+erlang_app_info(
+    name = "cowlib",
+    srcs = [
+        "include/cow_inline.hrl",
+        "include/cow_parse.hrl",
+        "src/cow_base64url.erl",
+        "src/cow_cookie.erl",
+        "src/cow_date.erl",
+        "src/cow_hpack.erl",
+        "src/cow_hpack_dec_huffman_lookup.hrl",
+        "src/cow_http.erl",
+        "src/cow_http2.erl",
+        "src/cow_http2_machine.erl",
+        "src/cow_http_hd.erl",
+        "src/cow_http_struct_hd.erl",
+        "src/cow_http_te.erl",
+        "src/cow_iolists.erl",
+        "src/cow_link.erl",
+        "src/cow_mimetypes.erl",
+        "src/cow_multipart.erl",
+        "src/cow_qs.erl",
+        "src/cow_spdy.erl",
+        "src/cow_spdy.hrl",
+        "src/cow_sse.erl",
+        "src/cow_uri.erl",
+        "src/cow_uri_template.erl",
+        "src/cow_ws.erl",
+        "src/cowlib.app.src",
+    ],
+    hdrs = [
+        "include/cow_inline.hrl",
+        "include/cow_parse.hrl",
+    ],
+    app = ":app_file",
+    app_name = "cowlib",
+    beam = [
+        "ebin/cow_base64url.beam",
+        "ebin/cow_cookie.beam",
+        "ebin/cow_date.beam",
+        "ebin/cow_hpack.beam",
+        "ebin/cow_http.beam",
+        "ebin/cow_http2.beam",
+        "ebin/cow_http2_machine.beam",
+        "ebin/cow_http_hd.beam",
+        "ebin/cow_http_struct_hd.beam",
+        "ebin/cow_http_te.beam",
+        "ebin/cow_iolists.beam",
+        "ebin/cow_link.beam",
+        "ebin/cow_mimetypes.beam",
+        "ebin/cow_multipart.beam",
+        "ebin/cow_qs.beam",
+        "ebin/cow_spdy.beam",
+        "ebin/cow_sse.beam",
+        "ebin/cow_uri.beam",
+        "ebin/cow_uri_template.beam",
+        "ebin/cow_ws.beam",
+    ],
+    license_files = ["LICENSE"],
+    visibility = ["//visibility:public"],
+)
+
+alias(
+    name = "erlang_app",
+    actual = ":cowlib",
+    visibility = ["//visibility:public"],
+)
