diff --git a/MODULE.bazel b/MODULE.bazel
index 184d526..b64f9ff 100644
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -1,6 +1,6 @@
 module(
-    name = "ra",
-    version = "2.4.0",
+    name = "rabbitmq_ra",
+    version = "2.6.3",
 )
 
 bazel_dep(
@@ -94,7 +94,7 @@ erlang_dev_package = use_extension(
 erlang_package.hex_package(
     name = "eunit_formatters",
     version = "0.5.0",
-    build_file = "@ra//bzl:BUILD.eunit_formatters",
+    build_file = "@rabbitmq_ra//bzl:BUILD.eunit_formatters",
 )
 
 use_repo(
