--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,72 @@
+load("@rules_erlang//:erlang_bytecode.bzl", "erlang_bytecode")
+load("@rules_erlang//:app_file.bzl", "app_file")
+load("@rules_erlang//:erlang_app_info.bzl", "erlang_app_info")
+load("@rules_erlang//:untar.bzl", "untar")
+
+untar(
+    name = "contents",
+    outs = [
+        "LICENSE.txt",
+        "README.md",
+        "rebar.config",
+        "rebar.lock",
+        "src/base64url.app.src",
+        "src/base64url.erl",
+    ],
+    archive = "contents.tar.gz",
+)
+
+erlang_bytecode(
+    name = "beam_files",
+    srcs = ["src/base64url.erl"],
+    hdrs = [],
+    dest = "ebin",
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": [
+            "+debug_info",
+            "+warn_format",
+            "+warn_export_vars",
+            "+warn_obsolete_guard",
+            "+warn_bif_clash",
+        ],
+        "//conditions:default": [
+            "+debug_info",
+            "+warn_format",
+            "+warn_export_vars",
+            "+warn_obsolete_guard",
+            "+warn_bif_clash",
+            "+deterministic",
+        ],
+    }),
+)
+
+app_file(
+    name = "app_file",
+    app_description = "URL safe base64-compatible codec",
+    app_name = "base64url",
+    app_src = ["src/base64url.app.src"],
+    app_version = "1.0.1",
+    dest = "ebin",
+    modules = [":beam_files"],
+    stamp = 0,
+)
+
+erlang_app_info(
+    name = "base64url",
+    srcs = [
+        "src/base64url.app.src",
+        "src/base64url.erl",
+    ],
+    hdrs = [],
+    app = ":app_file",
+    app_name = "base64url",
+    beam = [":beam_files"],
+    license_files = ["LICENSE.txt"],
+    visibility = ["//visibility:public"],
+)
+
+alias(
+    name = "erlang_app",
+    actual = ":base64url",
+    visibility = ["//visibility:public"],
+)
