--- /dev/null
+++ BUILD.bazel
@@ -0,0 +1,75 @@
+load("@rules_erlang//:untar.bzl", "untar")
+load("@rules_erlang//:app_file.bzl", "app_file")
+load("@rules_erlang//:erlang_bytecode.bzl", "erlang_bytecode")
+load("@rules_erlang//:erlang_app_info.bzl", "erlang_app_info")
+
+untar(
+    name = "extracted",
+    archive = "contents.tar.gz",
+    outs = [
+        "src/accept_header.erl",
+        "src/accept_neg.erl",
+        "src/accept_encoding_header.erl",
+        "src/accept_parser.erl",
+        "src/accept.app.src",
+        "include/accept.hrl",
+        "LICENSE",
+    ],
+)
+
+erlang_bytecode(
+    name = "beam_files",
+    hdrs = [
+        "include/accept.hrl",
+    ],
+    srcs = [
+        "src/accept_header.erl",
+        "src/accept_neg.erl",
+        "src/accept_encoding_header.erl",
+        "src/accept_parser.erl",
+    ],
+    erlc_opts = select({
+        "@rules_erlang//:debug_build": ["+debug_info"],
+        "//conditions:default": ["+deterministic", "+debug_info"],
+    }),
+    dest = "ebin",
+)
+
+app_file(
+    name = "app_file",
+    app_name = "accept",
+    app_version = "0.3.5",
+    app_description = "Accept header(s) for Erlang/Elixir.",
+    app_src = ["src/accept.app.src"],
+    modules = [":beam_files"],
+    dest = "ebin",
+    stamp = 0,
+)
+
+erlang_app_info(
+    name = "accept",
+    app_name = "accept",
+    hdrs = [
+        "include/accept.hrl",
+    ],
+    app = ":app_file",
+    beam = [":beam_files"],
+    license_files = [
+        "LICENSE",
+    ],
+    srcs = [
+        "src/accept_header.erl",
+        "src/accept_neg.erl",
+        "src/accept_encoding_header.erl",
+        "src/accept_parser.erl",
+        "src/accept.app.src",
+        "include/accept.hrl",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+alias(
+    name = "erlang_app",
+    actual = ":accept",
+    visibility = ["//visibility:public"],
+)
